CHANGELOG.md

```diff
@@ -79,15 +79,85 @@ Task.Supervisor.async(
 
 ## Improved errors on binaries and evaluation
 
-TODO.
+OTP 25 improved errors on binary construction and evaluation. These improvements
+apply to Elixir as well. Before v1.14, errors when constructing binaries would
+often be hard-to-debug generic "argument errors". With OTP 25 and Elixir v1.14,
+more detail is provided for easier debugging. This work is part of [EEP
+54](https://www.erlang.org/eeps/eep-0054).
+
+```elixir
+# Before:
+int = 1
+bin = "foo"
+int <> bin
+# ** (ArgumentError) argument error
+
+# Now:
+int = 1
+bin = "foo"
+int <> bin
+# ** (ArgumentError) construction of binary failed:
+# segment 1 of type 'binary':
+# expected a binary but got: 1
+```
 
 ## Slicing with steps
 
-TODO.
+Elixir v1.12 introduced **stepped ranges**, which are ranges where you can
+specify the "step":
+
+```elixir
+Enum.to_list(1..10//3)
+#=> [1, 4, 7, 10]
+```
+
+Stepped ranges are particularly useful for numerical operations involving
+vectors and matrices (see [Nx](https://github.com/elixir-nx/nx), for example).
+However, the Elixir standard library was not making use of stepped ranges in its
+APIs. Elixir v1.14 starts to take advantage of steps with support for stepped
+ranges in `Enum.slice/2`:
+
+```elixir
+letters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j"]
+Enum.slice(integers, 0..5//2)
+#=> ["a", "c", "e"]
+```
```

Comment:
Let's also add that `binary_slice/2` and `binary_slice/3` were added to Kernel, which are slice operations exclusive to binaries.

Reply:
I didn't add `binary_slice/3` in this section since it has nothing to do with stepped ranges and it's already mentioned in the sections below with the list of added functions. I added `binary_slice/2` though, great callout 🙃